packages/fiber/src/core/index.tsx

```diff
@@ -121,11 +122,33 @@ export type ReconcilerRoot<TCanvas extends Element> = {
   unmount: () => void
 }
 
+function computeInitialSize(canvas: HTMLCanvasElement, defaultSize?: Size): Size {
+  if (defaultSize) {
+    return defaultSize
+  }
+
+  if (canvas.parentElement) {
+    const { width, height, top, left } = canvas.parentElement.getBoundingClientRect()
+
+    return { width, height, top, left }
+  }
+
+  return { width: 0, height: 0, top: 0, left: 0 }
+}
+
+function elementIsCanvas(maybeCanvas: Element): maybeCanvas is HTMLCanvasElement {
+  return maybeCanvas.tagName === 'CANVAS'
+}
+
 function createRoot<TCanvas extends Element>(canvas: TCanvas): ReconcilerRoot<TCanvas> {
+  if (!elementIsCanvas(canvas)) {
```

Comment:
I may be missing context here. The generic `TCanvas` seems strange since it seems like this function wouldn't work if `canvas` is an arbitrary element other than canvas.

Reply:
Indeed. I don't believe `OffscreenCanvas` is too different, but I could be wrong. native also shims an `HTMLCanvasElement`.

Reply:
I've improved this to handle `OffscreenCanvas` explicitly and split this work out to hopefully allow the less controversial bug fix here to go ahead:
https://github.com/kmannislands/react-three-fiber/pull/2

I'll re-target this PR to r3f if there's appetite for the change.

Reply:
I'd have to defer to @joshuaellis on how to type this in https://github.com/kmannislands/react-three-fiber/pull/2. I'm a little wary of consuming types from `@types/three` rather than fixing upstream.